TCD-2831: Sauce-storage (#2314)
saucelabs/sauce-docs: docs/web-apps/automated-testing/selenium/pre-run-executables.md

### 64 v. 32-bit Version of AutoIT

The 64bit version of AutoIT works on IE11, and not on IE9. The 32bit version works with both browser versions.

## Automated Testing

### Upload Files with the REST API

Below are some examples of how to use the Sauce Labs REST API to upload your mobile file to Sauce Storage. If you do not have a file to test, consider using the [Sauce Labs Swag Labs sample app](https://github.com/saucelabs/sample-app-mobile) for validating this process.

#### REST API Authentication

> The authorization credentials are located here: [app.saucelabs.com](https://app.saucelabs.com/user-settings). A recommended best practice is to set your credentials as environment variables like so:

Reviewer: ```suggestion
You can find the authorization credentials at [app.saucelabs.com](https://app.saucelabs.com/user-settings). A recommended best practice is to set your credentials as environment variables, as shown below:
```

```bash
SAUCE_USERNAME='valid.username'
SAUCE_ACCESS_KEY='valid.key'
```

> For specific instructions on how to set environment variables visit, the following links:

Reviewer: ```suggestion
For specific instructions on how to set environment variables visit the following links:
```

- [Set Environment Variables with Windows 10](https://www.architectryan.com/2018/08/31/how-to-change-environment-variables-on-windows-10/)
- [Set Environment Variables with MacOS](https://apple.stackexchange.com/questions/106778/how-do-i-set-environment-variables-on-os-x)
- [Set Environment Variables with Linux](https://askubuntu.com/questions/58814/how-do-i-add-environment-variables)

#### Accepted File Types

> App Storage recognizes .apk files as Android apps and .ipa files as iOS apps, and will also parse a .zip file to determine whether a valid .app bundle exists, accepting it as an iOS app if it does. You can also upload and store other file types for generic use, such as a pre-run executable, package, or binary. Some of the formats for this type of use case include:

Reviewer: ```suggestion
App Storage identifies .apk files as Android apps and .ipa files as iOS apps. Additionally, it can analyze a .zip file to check for a valid .app bundle, which it will accept as an iOS app if found. You can also upload and store other file types for generic use, such as a pre-run executable, package, or binary. Some of the formats for this type of use case are as follows:
```

- .js
- .py
- .tar
- .zip
- .sh
- .bat

#### Team Management Sync

> App Storage uses a Team Management sync feature that allows for user permissions schemes. In other words, a Sauce Labs admin (either an org admin or a team admin) can control access to individual application files or specific binary/script files. By default, all uploaded files are shared with the same team where the user participates currently. You, as a user, can only access files that are shared with the team where you contribute/participate unless your role is an organization admin in which case you have access to all files in your particular organization.

Reviewer: ```suggestion
App Storage uses a Team Management sync feature that allows for user permissions schemes. In other words, a Sauce Labs admin (whether an organization admin or a team admin) can manage access to individual application files or specific binary/script files. By default, all uploaded files are shared with the same team to which the user currently belongs. As a user, you can only access files that are shared within your contributing/participating team, unless your role is an organization admin in which case you have access to all files in your particular organization.
```

> App Storage uses a Team Management sync feature that allows for user permissions schemes. In other words, a Sauce Labs admin (either an org admin or a team admin) can control access to individual application files or specific binary/script files. By default, all uploaded files are shared with the same team where the user participates currently. You, as a user, can only access files that are shared with the team where you contribute/participate unless your role is an organization admin in which case you have access to all files in your particular organization.

Author: ```suggestion
Corrected sentence:

App Storage uses a Team Management sync feature that allows user permission schemes. In other words, a Sauce Labs admin (either an org admin or a team admin) can control access to individual application files or specific binary/script files. By default, all uploaded files are shared with the same team in which the user participates. As a user, you can only access files shared with the team where you contribute/participate unless your role is an organization admin; in this case, you have access to all files in your particular organization.
```
